mdr/mdr.py

```diff
@@ -57,6 +61,19 @@ def fit(self, features, classes):
         None
 
         """
+        self.class_ratio = float(sum(classes == 0))/(classes.size) #only applies to binary classification 
+        num_classes = (np.unique(classes)).size # count all the unique values of classes 
+        self.feature_map = defaultdict(lambda: np.zeros((num_classes,), dtype=np.int))
+
+        for row_i in range(features.shape[0]):
+            feature_instance = tuple(map(tuple, features[row_i])) #convert feature vector to tuple 
+            feature_map[feature_instance][classes[row_i]] += 1 #update count 
+
+        
+
```

Comment:
`feature_map` should not be the counts. Use a variable called `class_count_matrix` instead.

Once you're done counting in `fit`, go through each tuple and calculate the class fraction for each tuple:

1) If both classes for that tuple are 0, then use `self.default_label`
2) If the class fraction == `self.class_ratio`, then use `self.tie_break`
3) If the class fraction is > `self.class_ratio`, then it is the first class
4) If the class fraction is < `self.class_ratio`, then it is the second class